### PR TITLE
Diksha prod bug fix

### PR DESCRIPTION
**Pull request by ankitpws**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: migrations/updateUserProfileAndMissMatchOfRoleInformation/updateProjectWithProfileData.js [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+                },
+                update: {
+                  $set: {
+                    userProfile: userProfile,
+                  },
+                },
+              },
+            };
+
+            updateProjectWithuUserProfile.push(updateObject);
```
Comment (aks30) [inline migrations/updateUserProfileAndMissMatchOfRoleInformation/updateProjectWithProfileData.js:288]: @ankitpws - Fix spelling mistake

File: release-notes/7.0.0.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    1. Log in to Jenkins.
+    2. Navigate to Dashboard -> AutoBuild -> StagingManual -> managed-learn -> ml-projects-service. OR for dev Navigate to Dashboard -> Build -> managed-learn -> ml-projects-service
+    3. Click on "Build with parameters" and provide the latest release tag in the field labeled "github_release_tag". Initiate the build process.
+    4. The build job will take approximately 5 minutes to complete.
+    5. After the job finishes, go to Dashboard -> Deploy -> staging -> managed-learn -> ml-projects-service. OR for dev go to Dashboard -> Deploy -> dev -> managed-learn -> ml-projects-service This job will be executed automatically in the dev environment. If not, then it should be built manually.
+    6. Click on "Build with parameters" to initiate the deployment process.
+    7. Once the job is completed, the services will be deployed on the staging environment
+
+## Migrations
```
Comment (aks30) [inline release-notes/7.0.0.md:19]: @ankitpws - Mention that this release contains certain other migrations and link the readme for the same.

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 9 Nov

File: migrations/updateDistrictNameInProjects/Readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+#### Steps to run the script files
+
+In order to execute this migration script, we need to first log in to the pod where the service is running and then proceed with the provided instructions.
+
+This script is designed to update project documents specifically for cases where there is a mismatch in district names. It will only affect the IDs specified in the script.
+
+### Step 1:
+
+    Navigate to /opt/projects/updateDistrictNameInProjects/
```
[the remark this answers is not in the document]
Comment (ankitpws) [reply migrations/updateDistrictNameInProjects/Readme.md]: added ticket and updated migration as well

File: migrations/updateUserProfileAndMissMatchOfRoleInformation/Readme.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+#### Steps to run the script files
+
+In order to execute this migration script, we need to first log in to the pod where the service is running and then proceed with the provided instructions.
+
+This script will update projects in cases where the user profile is missing and there is a mismatch between userProfile and userRoleInformations.
+
+### Step 1:
+
+    Navigate to /opt/projects/updateUserProfileAndMissMatchOfRoleInformation/
```
[the remark this answers is not in the document]
Comment (ankitpws) [reply migrations/updateUserProfileAndMissMatchOfRoleInformation/Readme.md]: added ticket and updated migration as well

File: migrations/updateUserProfileAndMissMatchOfRoleInformation/updateProjectWithProfileData.js (hunk 1 above)
Comment (ankitpws) [reply migrations/updateUserProfileAndMissMatchOfRoleInformation/updateProjectWithProfileData.js]: fixed updateProjectWithuUserProfile to updateProjectWithUserProfile

File: release-notes/7.0.0.md (hunk 2 above)
Comment (ankitpws) [reply release-notes/7.0.0.md]: removed as it will go in 6.0.0 release

File: migrations/updateDistrictNameInProjects/Readme.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+### Step 1:
+
+    Navigate to /opt/projects/migrations/updateDistrictNameInProjects/
+
+### Step 2:
+
+Run the script to update projects.
+
+    node updateUserProfileDistrictNameMissing.js
```
Comment (aks30) [inline migrations/updateDistrictNameInProjects/Readme.md:35]: @ankitpws Let's add verification steps here, meaning after each script some records can be verified to confirm if the scripts have executed successfully.

File: migrations/updateDistrictNameInProjects/Readme.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    node updateDistrictName.js
+
+This script is designed to refresh project documents by incorporating the most recent profile information along with the individual's previous role in the project.
+
+### Step 1:
+
+    Navigate to /opt/projects/migrations/updateDistrictNameInProjects/
+
+### Step 2:
```
Comment (aks30) [inline migrations/updateDistrictNameInProjects/Readme.md:31]: @ankitpws we should add step 3 for updateDistrictName.js as well right

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 15 Nov

File: migrations/updateDistrictNameInProjects/Readme.md (hunk 5 above)
Comment (ankitpws) [reply migrations/updateDistrictNameInProjects/Readme.md:35]: like they can use DB find api to validate? @aks30

**Pull request merged**